src/libtcod/renderer_xterm.c

```diff
@@ -159,7 +173,138 @@ static void xterm_destructor(struct TCOD_Context* __restrict self) {
   xterm_cleanup();
 }
 
-TCOD_Context* TCOD_renderer_init_xterm(const char* window_title) {
+static void xterm_handle_input_escape() {
+  if (getchar() != '[')
+    return;
+  char rows_buf[16] = "";
+  char cols_buf[16] = "";
+  for (int i = 0; i < 15; i++) {
+    int ch = getchar();
+    if (ch == ';') {
+      rows_buf[i] = '\0';
+      break;
+    }
+    if (!isdigit(ch)) {
+      return;
+    }
+    rows_buf[i] = ch;
+  }
+  for (int j = 0; j < 15; j++) {
+    int ch = getchar();
+    if (ch == 'R') {
+      cols_buf[j] = '\0';
+      break;
+    }
+    if (!isdigit(ch)) {
+      return;
+    }
+    cols_buf[j] = ch;
+  }
+  g_got_rows = atoi(rows_buf);
+  g_got_columns = atoi(cols_buf);
+  g_got_size_timestamp = SDL_GetTicks();
+}
+
+static int xterm_handle_input(void *arg) {
+  while (true) {
+    int ch = getchar();
+    if (ch == '\x1b') {
+      xterm_handle_input_escape();
+      continue;
+    }
+    int sym = ch;
+    int mod = KMOD_NONE;
+    if isupper(ch) {
```

Comment:
Missing parentheses here.
```suggestion
    if (isupper(ch)) {
```